mantis-connectors/mantis-connector-publish/src/main/java/io/mantisrx/connector/publish/source/http/SourceSink.java

```diff
@@ -46,25 +45,30 @@ public Void call(Map<String, List<String>> t1, Context t2) {
 
     public SourceSink(Func2<Map<String, List<String>>, Context, Void> preProcessor,
                       Func2<Map<String, List<String>>, Context, Void> postProcessor, String mantisClientId) {
-        this.postProcessor = postProcessor;
-        this.preProcessor = preProcessor;
-        this.clientId = mantisClientId;
+        this.sink = new ServerSentEventsSink.Builder<String>()
+            .withEncoder(data -> data)
+            .withPredicate(new Predicate<>("description", new EventFilter(mantisClientId)))
+            .withRequestPreprocessor(preProcessor)
+            .withRequestPostprocessor(postProcessor)
+            .build();
     }
 
     @Override
     public void call(Context context, PortRequest portRequest,
                      Observable<String> observable) {
         observable = observable.filter(t1 -> !t1.isEmpty());
 
-        ServerSentEventsSink<String> sink = new ServerSentEventsSink.Builder<String>()
-                .withEncoder(data -> data)
-                .withPredicate(new Predicate<>("description", new EventFilter(clientId)))
-                .withRequestPreprocessor(preProcessor)
-                .withRequestPostprocessor(postProcessor)
-                .build();
-
-        observable.subscribe();
+        subscription = observable.subscribe();
 
         sink.call(context, portRequest, observable);
     }
+
+    @Override
+    public void close() throws IOException {
+        try {
+            sink.close();
+        } finally {
+            subscription.unsubscribe();
+        }
```

Comment:
@calvin681 should we unsubscribe before doing the close?

Reply:
TBH, I'm not even sure why we need to eager subscribe to the observable in `call`. Regardless, unsubscribing first would make more logical sense.